sigstore/_verify.py

```diff
@@ -249,53 +249,40 @@ def verify(
         )
 
         # Retrieve the relevant Rekor entry to verify the inclusion proof and SET
-        uuids = self._rekor.index.retrieve.post(sha256_artifact_hash, pub_b64.decode())
+        entry = self._rekor.log.entries.retrieve.post(
```

Comment:
Experimentally it looks like the endpoint returns `404` instead of an empty list, but I'll add the check for resiliency. 

(I'm also going to update the Python API here to return an `Optional[RekorEntry]`, since that's much more ergonomic than tracking an HTTP-sourced exception that could be either `404` or something else.)

Reply:
https://github.com/sigstore/sigstore-python/pull/238/files#diff-45c49b779739a3e08051f2c6afe622eb6b693d600dcc0ad9052ac30139a19a62R261-R267